include/cudaq/Optimizer/Dialect/Quake/QuakeOps.td

```diff
@@ -586,14 +586,34 @@ def quake_SinkOp : QuakeOp<"sink"> {
   let assemblyFormat = [{
      $target `:` qualified(type(operands)) attr-dict
   }];
+
+  let extraClassDeclaration = [{
+    //===------------------------------------------------------------------===//
+    // WireInterface
+    //===------------------------------------------------------------------===//
+
+    void setWireOperands(mlir::ValueRange wires) {
+      assert(wires.size() == 1 && "SinkOp has only one wire operand");
+      getTargetMutable().assign(wires[0]);
+    }
+
+    mlir::ValueRange getWireOperands() {
+      return getOperation()->getOperands();
+    }
+
+    mlir::ValueRange getWires() {
+      return {};
+    }
```

Comment:
@schweitzpgi - the `getWires()` function name is also used in `OperatorInterface` and `MeasurementInterface`. In those cases, the description associated with the function is: `Returns the set of qubits that were operated upon`. I'm a bit concerned that with this change, the `sink` operation would now be uniquely overriding that to mean something different. That is - the `sink` operation's meaning of `getWires()` is essentially `Returns the set of output quantum states`. Do you think that is a potential problem?

Reply:
Yes. It is inconsistent.

For Operators, getWires() results the results of the Op. For Measurements, it returns the operands. Meanwhile setWires() in the new interface sets all the control and target operands without checking their types or the type of the results (if any).

I'm concerned it is too inconsistent. Perhaps a simple function that iterates over the operands of a "generic" Op, using the standard interfaces, and correctly transforms operands of type `!wire` while asserting that arity is equal to coarity would be cleaner here.

Reply:
> Perhaps a simple function that iterates over the operands of a "generic" Op, using the standard interfaces, and correctly transforms operands of type !wire while asserting that arity is equal to coarity would be cleaner here.

Updated in [f7406fc](https://github.com/NVIDIA/cuda-quantum/pull/666/commits/f7406fc6335c4346d1c2c56db04a05effb2d5595). Let me know if this is what you had in mind.